### ISSUE TITLE
Refactor Channel - introduce Observable.unicast and Observable.broadcast

### ISSUE THREAD

**Issue by alexandru** [monixio/monix#98]
`Channel` is a good concept. First, lets remember that `Channel` has this interface:

```scala
trait Channel[-I] { self =>
  def pushNext(elem: I): Unit
  def pushComplete(): Unit
  def pushError(ex: Throwable): Unit
}
```

We then provide various channel types, such as the `PublishChannel`, the `BehaviorChannel`, the `AsyncChannel` and the `ReplayChannel` (backed by corresponding Subjects). Unfortunately I've noticed that in practice these channels are being misused, much like Subjects are in Rx.

An alternative would be to provide a new builder like this (inspired by the Enumerators in Play):

```scala
object Observable {
  def unicast[T](f: Channel[T] => Unit): Observable[T] = ???
}
```

Such a builder would take care of many use-cases while being safer to use. It would also have speed benefits, because our `Channel` instance (passed in that function) only needs to accommodate a single subscriber. This means that we do not need a backing subject.

And then to accommodate more diverse use-cases, like say we really need a hot/shared channel, we should have another builder like this:

```scala
object Observable {
  def broadcast[T](os: OverflowStrategy): (Channel[T], Observable[T])
  def broadcast[I,O](subject: Subject[I,O], os: OverflowStrategy): (Channel[T], Observable[T])
}
```

In other words:

- no more nonsense like `ObservableChannel` (e.g. a supplied interface for an Observable that's also a channel), enough of that
- no more `PublishChannel`, `BehaviorChannel`, `AsyncChannel`, `ReplayChannel` or any of that stuff: if you want the same behavior, you are forced to work with `Observable.unsafeBroadcast(OverflowStrategy)`, in which case OverflowStrategy will still be mandatory, like it is right now, to force a choice and to make it clear how unsafe it is
- it's usage should be slightly less convenient, *because* you get a tuple instead of getting an instance of something that's both a `Channel` and an `Observable` (like `PublishChannel` is)

**Comment by alexandru** [comment]
Ticket is no longer relevant. Rejecting.
